src/os/bluestore/BlueStore.cc

```diff
@@ -6487,29 +6494,39 @@ int BlueStore::_open_db(bool create, bool to_repair_db, bool read_only)
   }
   if (r) {
     derr << __func__ << " erroring opening db: " << err.str() << dendl;
-    _close_db(read_only);
+    _close_db();
     return -EIO;
   }
   dout(1) << __func__ << " opened " << kv_backend
 	  << " path " << kv_dir_fn << " options " << options << dendl;
   return 0;
 }
 
-void BlueStore::_close_db(bool cold_close)
+void BlueStore::_close_db_leave_bluefs()
 {
   ceph_assert(db);
   delete db;
-  db = NULL;
-  if (bluefs) {
-    _close_bluefs(cold_close);
-  }
+  db = nullptr;
 }
 
-void BlueStore::_close_db_leave_bluefs()
+void BlueStore::_close_db()
 {
-  ceph_assert(db);
-  delete db;
-  db = nullptr;
+  dout(10) << __func__ << "::NCB::db_was_opened_read_only=" << db_was_opened_read_only << dendl;
+  dout(10) << __func__ << "::NCB::fm=" << fm << " need_to_destage_allocation_file=" << need_to_destage_allocation_file << dendl;
```

Comment:
IMO one can use a single log line to dump all the desired info. And I think more compact naming (e.g. use was_readonly and need_destage instead of full var names) would be better.

Reply:
There is no reason to spam BlueStore logs with NCB (no-column-b) markers in places that are not directly related to this feature.